spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/ReactorSleuth.java

```diff
@@ -67,43 +66,37 @@ private ReactorSleuth() {
 			log.trace("Scope passing operator [" + beanFactory + "]");
 		}
 
-		// Adapt if lazy bean factory
-		BooleanSupplier isActive = beanFactory instanceof ConfigurableApplicationContext
-				? ((ConfigurableApplicationContext) beanFactory)::isActive : () -> true;
-
 		return Operators.liftPublisher((p, sub) -> {
-			// if Flux/Mono #just, #empty, #error
+			// While supply of scalar types may be deferred, we don't currently scope
```

Comment:
Side note: the contract expressed in the `ScalarCallable` javadoc should be made more explicit, but yeah a `ScalarCallable` should resolve immediately without latency, to a terminating `Publisher`.

There's not a lot of them in core: just, empty, error, and errorSupplied. The only outlier is `errorSupplied`, since the Throwable is not pre-computed but lazily generated. But that's still considered an instant termination for all intent and purposes.

Reply:
thanks.. updating in the next PR!